src/transformers/models/modernbert/modular_modernbert.py

```diff
@@ -1428,6 +1280,39 @@ def forward(
         )
 
 
+def cls_pooling(hidden_states: torch.Tensor, attention_mask: torch.Tensor) -> torch.Tensor:
+    return hidden_states[:, 0]
+
+
+def mean_pooling(hidden_states: torch.Tensor, attention_mask: torch.Tensor) -> torch.Tensor:
+    return (hidden_states * attention_mask.unsqueeze(-1)).sum(dim=1) / attention_mask.sum(dim=1, keepdim=True)
+
+
+MODERNBERT_POOLING_FUNCTION = {
+    "cls": cls_pooling,
+    "mean": mean_pooling,
+}
+
```

Comment:
as per our previous offline discussino let's only add one Head, with cLS if it's the one used, and no one liner for the pooling!

Reply:
Tackled as discussed offline in 92e17c667aae8e9473237e1ec9a843f16eb58e70